src/frontend/src/statement.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod describe;
+mod show;
+mod tql;
+
+use catalog::CatalogManagerRef;
+use common_query::Output;
+use common_recordbatch::RecordBatches;
+use query::parser::QueryStatement;
+use query::query_engine::StatementExecutorExtRef;
+use query::QueryEngineRef;
+use session::context::QueryContextRef;
+use snafu::{ensure, ResultExt};
+use sql::statements::statement::Statement;
+
+use crate::error::{
+    CatalogSnafu, ExecLogicalPlanSnafu, ExecuteStatementSnafu, PlanStatementSnafu, Result,
+    SchemaNotFoundSnafu,
+};
+
+#[derive(Clone)]
+pub(crate) struct StatementExecutor {
+    catalog_manager: CatalogManagerRef,
+    query_engine: QueryEngineRef,
+    executor_ext: StatementExecutorExtRef,
+}
+
+impl StatementExecutor {
+    pub(crate) fn new(
+        catalog_manager: CatalogManagerRef,
+        query_engine: QueryEngineRef,
+        executor_ext: StatementExecutorExtRef,
+    ) -> Self {
+        Self {
+            catalog_manager,
+            query_engine,
+            executor_ext,
+        }
+    }
+
+    pub(crate) async fn execute_sql(
```

Comment:
why not use `execute_stmt`? Its name is `StatementExecutor`, but the main function name is `execute_sql`.

Reply:
+1 for `execute_stmt`. And accepts `QueryStatement` enum rather than the `Statement` special for SQL.